### PR TITLE
Facilitator survey results use new roll-up questions

### PR DESCRIPTION
**Pull request by clareconstantine**
[PLC-286](https://codedotorg.atlassian.net/browse/PLC-286)
[Spec](https://docs.google.com/document/d/1MobZVPFrKg2SYO3SMEuP0X3ndnQD86ae4xDtZinmPc4/edit?ts=5cedbf45#) - although the facilitator survey itself, not this spec, is now the source of truth.

Screen Shot 2019-05-30 at 3 51 05 PM

### CODE REVIEW

**Review by agealy — APPROVED** [review]
For posterity, I believe the duplicated columns is an existing issue that we're not worrying about for now and can address for all affected dashboards in a future update.

Does the behavior of those columns change depending on user permissions, or is it dependent on the existence of other workshops hosted by that facilitator?

**Comment by clareconstantine** [comment]
> For posterity, I believe the duplicated columns is an existing issue that we're not worrying about for now and can address for all affected dashboards in a future update.
> 
> Does the behavior of those columns change depending on user permissions, or is it dependent on the existence of other workshops hosted by that facilitator?

@agealy it varies - the way it looks to people with other permissions is documented here: https://docs.google.com/document/d/1_1FtOKf-1A2zu6C8TjWzcjKWPnstP0NFeG57K79lzP8/edit

we only see the duplicate for workshop admins.

File: dashboard/test/helpers/pd/workshop_survey_results_helper_test.rb [hunk 1]
```diff
@@ -759,61 +759,63 @@ class Pd::WorkshopSurveyResultsHelperTest < ActionView::TestCase
...
           all_my_workshops: nil
         },
-        'iFeel133' => {
-          this_workshop: 4.63,
+        'overall_success_0' => {
+          this_workshop: 4.0,
           all_my_workshops: nil
         },
+        # these values come from the average response to all questions in the category
+        # so, 4 / (category size)
```
Comment (hacodeorg) [inline dashboard/test/helpers/pd/workshop_survey_results_helper_test.rb:818]: ahhh that's why!

File: dashboard/app/helpers/pd/workshop_survey_results_helper.rb [hunk 2]
```diff
@@ -427,9 +429,9 @@ def generate_facilitator_averages(summary)
     facilitator_averages = facilitators.map {|name| [name, {}]}.to_h
     facilitator_averages[:questions] = {}
 
-    # In some cases, the same question has different IDs in different forms. To get around
+    # In some cases, the same question has different question names in different forms. To get around
     # this, the QUESTIONS_FOR_FACILITATOR_AVERAGES contains a primary_id (what we use
-    # here) and optional all_ids (what is used in multiple forms)
+    # here) and optional all_ids (names used in multiple forms)
```
Comment (hacodeorg) [inline dashboard/app/helpers/pd/workshop_survey_results_helper.rb:434]: Not specific to this PR, now we have a chance to re-do this facilitator survey, we should enforce a specific set of question names.

**Comment by hacodeorg** [comment]
Looks good to me. IMO, there are so many hardcoded things in the UI layer that we should find a solution for later.

**Review by hacodeorg — APPROVED** [review]

**Pull request merged**